### ISSUE TITLE
Grid height not resizes properly in some cases.

### ISSUE THREAD

**Issue by gokulsaravanan** [xamarin/Xamarin.Forms#12557]
### Description

When an editor is loaded inside a Grid with column width star and another column as Auto. When the text changes in the editor, editor height resizes as excepted and but its parent(Grid) height is not resized properly. It resizes after typing some more text. 

The issue occurs in Android, iOS, and UWP

### Steps to Reproduce

1. Run the attached sample.
2. Type some text in the editor to multiple lines.
3. Notice the height resizes for an editor and the button(Red color). But its parent(Grid) view is not resized. 

`                <Grid Grid.Row="1" x:Name="typingAreaGrid" ColumnSpacing="0" BackgroundColor="Green" Padding="10">

                <Grid.RowDefinitions>
                    <RowDefinition Height="Auto"/>
                </Grid.RowDefinitions>

                <Grid.ColumnDefinitions>
                    <ColumnDefinition Width="*"/>
                    <ColumnDefinition Width="Auto"/>
                   <ColumnDefinition Width="80"/>
                </Grid.ColumnDefinitions>

                <Editor x:Name="typingArea" Placeholder="Type a message"
                        x:FieldModifier="Internal"
                        AutoSize="TextChanges"
                        TextChanged="typingArea_TextChanged"
                        Grid.Column="0" >
                </Editor>
                <Button BackgroundColor="Red" Text="send" TextColor="White" Grid.Column="1" Clicked="OnSendIconTapped" WidthRequest="100" />
                <BoxView BackgroundColor="Blue" Grid.Column="2"/>
            </Grid>
`
### Expected Behavior
Grid height needs to resize properly.

### Actual Behavior
The grid in which the editor is loaded is not resized properly when a column with the column definition as `Auto`  for the width.

### Basic Information

- Version with issue: Checked in 4.5.0.617. It also occurs in latest 4.8 stable version. 
- IDE: VS 2019
- Platform Target Frameworks: 
  - iOS:  13.5
  - Android: 9.0
- Nuget Packages: Xamarin.Forms

### Screenshots
![Screen Recording 2020-10-22 at 12 35 56 AM](https://user-images.githubusercontent.com/46105032/96773061-a4f56300-1401-11eb-8232-8559a2d25d69.gif)

[Issue video.zip](https://github.com/xamarin/Xamarin.Forms/files/5418206/Issue.video.zip)

### Reproduction Link

[EditorInGrid.zip](https://github.com/xamarin/Xamarin.Forms/files/5418192/EditorInGrid.zip)

> Issue occurs in Android, iOS, and UWP

**Comment by jsuarezruiz** [comment]
cc @hartez

**Comment by PureWeen** [comment]
@gokulsaravanan I'm not quite following your expectations here

Your outer most grid is set to 

```XAML
<Grid.RowDefinitions>
                <RowDefinition Height="*"/>
                <RowDefinition Height="*"/>
            </Grid.RowDefinitions>
```

That's telling each row to take up an equal percentage so the first child grid will take up 50 percent and the bottom label will take up 50 percent.

And then the inner grid is

```XAML
<Grid.RowDefinitions>
                    <RowDefinition Height="*"/>
                    <RowDefinition Height="Auto"/>
                </Grid.RowDefinitions>
```

Which seems consistent with the behavior I'm seeing. The Editor is part of that Auto so it keeps remeasuring as it grows and shrinks the ListView Row

**Comment by gokulsaravanan** [comment]
@PureWeen , Thanks for your attention.
Here I m not talking about ListView Row. The inner grid has two rows with below row properties

```
<Grid.RowDefinitions>
 <RowDefinition Height="*"/>   
 <RowDefinition Height="Auto"/> 
</Grid.RowDefinitions>

```
In this listview loaded in first row, in these i have no problem.

I am talking about the grid loaded in second row, in which as Editor and button is loaded.

```
 <Grid.ColumnDefinitions>
                <ColumnDefinition Width="*"/>
                <ColumnDefinition Width="Auto"/>
               <ColumnDefinition Width="80"/>
            </Grid.ColumnDefinitions>

            <Editor x:Name="typingArea" Placeholder="Type a message"
                    x:FieldModifier="Internal"
                    AutoSize="TextChanges"
                    TextChanged="typingArea_TextChanged"
                    Grid.Column="0" >
            </Editor>
            <Button BackgroundColor="Red" Text="send" TextColor="White" Grid.Column="1" Clicked="OnSendIconTapped" WidthRequest="100" />
            <BoxView BackgroundColor="Blue" Grid.Column="2"/>

```

Whenever the editor resizes the Grid with BackgroundColor as Green as shown in video not resize at instant when editor text enters the second line. It resizes after some text entered. It is clearly shown in the video that I have attached.

**Note:** For better understanding the issue increase the WidthRequest for the button which is loaded in the next column of editor. You can see the issue clearly. 

> The issue occurs only if the column definition is **Auto**. Here button is loaded in the column with auto width.

**Comment by jsuarezruiz** [comment]
@gokulsaravanan Testing the issue, seems to be fixed in the latest stable 4.8 version and  5.0-pre4. Could you try it?

![issue12557](https://user-images.githubusercontent.com/6755973/99534646-06c8ce80-29a8-11eb-9f2c-022c1169cd4d.gif)

The PR that seems to fix the issue: https://github.com/xamarin/Xamarin.Forms/pull/12475

**Comment by Redth** [comment]
Since we haven't heard from you in more than 30 days, we hope this issue is no longer affecting you. If it is, please reopen this issue and provide the requested information so that we can look into it further. Thank you!
